hcloud/resource_hcloud_reversedns.go

```diff
@@ -0,0 +1,241 @@
+package hcloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+func resourceReverseDNS() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceReverseDNSRead,
+		Create: resourceReverseDNSCreate,
+		Delete: resourceReverseDNSDelete,
+
+		Schema: map[string]*schema.Schema{
+			"server_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"floating_ip_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ip_address": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
`ip_address` could be made optional by defaulting to the public IPv4 address or the first address of the IPv6 subnet.
What are your thoughts on this?

I have not made up my mind yet, but this seems to be a nice shortcut for the majority use case.

Reply:
scratch this idea
Defaulting can only work with a single IP address or we cannot create a meaningful unique id for the terraform record. And for a only the ipv4, I doubt it is worth the hassle.

We can still add defaulting later, because it would not break existing configs.